mypy/semanal.py

```diff
@@ -6505,7 +6505,9 @@ def _lookup(
                 implicit_node = node
         # 2b. Class attributes __qualname__ and __module__
         if self.type and not self.is_func_scope() and name in {"__qualname__", "__module__"}:
-            return SymbolTableNode(MDEF, Var(name, self.str_type()))
+            v = Var(name, self.str_type())
+            v._fullname = name
```

Comment:
I think something like this would be more correct?
```suggestion
            v._fullname = self.qualified_name(name)
```

Reply:
I am going to try this myself to speed things up.